packages/expo/src/Expo.fx.tsx

```diff
@@ -18,3 +19,8 @@ if (isRunningInExpoGo()) {
   const globalHandler = ErrorUtils.getGlobalHandler();
   ErrorUtils.setGlobalHandler(createErrorHandler(globalHandler));
 }
+
+if (process.env.NODE_ENV !== 'production') {
```

Comment:
Maybe we could use `__DEV__` instead? Isn't it stripped by Metro?

Reply:
I believe statically analyzable references to `process.env.XX` are replaced by Metro (inlined constants). @byCedric and @EvanBacon can confirm whether to use `NODE_ENV` or `__DEV__` here.

Reply:
yes, metro inline plugin will do the replacement: https://github.com/facebook/metro/blob/main/packages/metro-transform-plugins/src/__tests__/inline-plugin-test.js

i choose NODE_ENV here because we have more NODE_ENV usage than __DEV__ in the `expo` package  😂